### DISCUSSION TITLE
Gerrit change is rebased all the time

### DISCUSSION THREAD

**Discussion by kasium** [renovatebot/renovate#28791] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gerrit

### Please tell us more about your question or problem

I have the following situation: a Gerrit change is open about a dependency update. The current master branch receives a new commit in the meantime. Renovate is configured to not rebase (rebase when conflicted), but still pushed a new patch set.

In the logs I can see `Rebasing branch after file not found`. As far as I understand the code, the getFile function tries to use the current branch, but this does not work, since Gerrit does not use branches.

This makes Renovate quite unusable since open changes are constantly rebased.

### Logs (if relevant)

<details><summary>Logs</summary>
```

DEBUG: syncBranchState() (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: syncBranchState(): Branch cache not found, creating minimal branchState (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: branchExists=true (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: dependencyDashboardCheck=undefined (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: PR rebase requested=false (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: Checking if PR has been edited (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: Found existing branch PR (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: Checking schedule(at any time, null) (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: No schedule defined (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: Branch already exists (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: Skipping behind base branch check due to rebaseWhen=conflicted (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: Branch does not need rebasing (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: Using reuseExistingBranch: true (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: Setting current branch to master (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: latest commit (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
       "branchName": "master",
       "latestCommitDate": "2024-05-02T12:52:20+00:00"
DEBUG: manager.getUpdatedPackageFiles() reuseExistingBranch=true (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: Rebasing branch after file not found (repository=renovate-test, packageFile=setup.py, branch=renovate/boto3-stubs-1.x)
       "depName": "boto3-stubs"
DEBUG: manager.getUpdatedPackageFiles() reuseExistingBranch=false (repository=renovate-test, branch=renovate/boto3-stubs-1.x)
DEBUG: Starting search at index 1624 (repository=renovate-test, packageFile=setup.py, branch=renovate/boto3-stubs-1.x)
       "depName": "boto3-stubs"
```
</details>

**Comment by felipecrs** [comment]
I had not noticed this behavior, but it's probably happening indeed.

Some ideas to fix it:

1. Create the branch locally and clone the change to it before letting Renovate compare
2. Use Gerrit APIs to detect the conflict.

**Reply by kasium** [reply]
@felipecrs not sure if you have the right, but can you please create an issue out of this?

**Reply by felipecrs** [reply]
Sorry, I don't have rights. I'm just an user like you :)

**Reply by RahulGautamSingh** [reply]
Gerrit support is experimental, so we do not prioritize it for the time being. Issues can be created once we declare it stable. We will keep this discussion in mind, nonetheless.

**Reply by kasium** [reply]
thanks for the update. Do you have a timeline until you declare it stable or let me ask otherwise, what are the guidelines to declare it as stable

**Reply by RahulGautamSingh** [reply]
We do not have any timeline or guideline, currently

**Reply by kasium** [reply]
okay, thank you

**Comment by kasium** [comment]
Would it be possible to create a bug out of this?
